include/data_struct.h

```diff
@@ -197,6 +217,13 @@ class BaseLSTMStates {
     void reset_zeros();
 };
 
+// Smoother for linear layer when using LSTM
+class BaseSLinear {
```

Comment:
what is different between `BaseSLinear` and `BaseTempSLinear`. It is too complicated and the codebase will be bloated soon. My rule of thumb is that we need to declare a class for a data structure if it is absolutely needed

Reply:
(1)`BaseTempSLinear` is a temporary data struct using in the forward pass to transfer some variables from `SLSTM` to `SLinear` layers.  (2) `BaseSLinear` is to saved variables needed for doing smoother in `SLinear`. (3) If I combine these two classes, I will need to modify `input_z_buffer` (i.e. modify `BaseHiddenStates`) to include the variables that I want to transfer `SLSTM` to `SLinear`. See above reply, what do you suggest: option 1 modify `BaseTempStates ` or option 2 modify `BaseHiddenStates`?

Reply:
See the comment above